### ISSUE TITLE
set-commits fails if any registered repo is invalid

### ISSUE THREAD

**Issue by TuurDutoit** [getsentry/sentry-cli#848]
## The issue
A few days ago, we suddenly started seeing some failures on our CircleCI workflows. This turned out to be caused by the very last step, which runs the following command (some parts censored):

```
sentry-cli --auth-token **** releases set-commits --auto com.****@****+****
```

and started returning the following error:

```
error: For non git repositories explicit revisions are required
```

However, this command *is* run within a Git repo, as it has been for months.

## Cause
Listing all our repos works as expected, although there is a weird entry at the end of that list:

| Name | Provider | URL |
|-|-|-|
| origin | Unknown Provider | - |

This entry does seem to be the reason for the error; I'm not really sure how that got in there, I'm asking my colleagues for clarification.

## Root cause
I did find it a little weird that this entry made sentry-cli spit out this error, so I dug a little deeper and found the actual error in the code:
https://github.com/getsentry/sentry-cli/blob/e6d8030864706781920dd74cb8936b017a5d617a/src/utils/vcs.rs#L233
If I understand this correctly, when running `sentry-cli releases set-commits --auto`, it loops through all registered repos to find out which one to use. However, when it encounters this weird entry, it bails out with this cryptic error, *even though a previous entry matched successfully*.
In short, the addition of this broken repo entry makes the commit association error out, even though the correct repo is still in there. The fact that repos are configured org-wide also means that *other* teams can (and did) break our team's CI...

**Comment by ehyland** [comment]
Thanks for reporting this @TuurDutoit, saved me a lot of time :) 

I just had the same issue, found a repo named origin with unknown provider in sentry settings. No idea how it got there.

https://sentry.io/settings/<org-name>/repos/

Deleting it fixed the issue for me

**Comment by kamilogorek** [comment]
@TuurDutoit @ehyland what version the the cli are you using? This issue _should_ have been already fixed 🤔 

https://github.com/getsentry/sentry-cli/blob/7d0470f98f6bcc67ec4bca5bc8c43be9fd01d7d4/src/utils/vcs.rs#L248-L252
https://github.com/getsentry/sentry-cli/blob/7d0470f98f6bcc67ec4bca5bc8c43be9fd01d7d4/src/utils/vcs.rs#L277-L279

(it should skip invalid entries now)

**Comment by TuurDutoit** [comment]
We're currently using `sentry/cli@1.53.0`; not sure why I didn't look at the code on master... I'll try updating!

**Comment by TuurDutoit** [comment]
@kamilogorek It seems to be working now with `@sentry/cli@1.59.0`!

**Comment by kamilogorek** [comment]
Thanks for verifying!
